server/transport.go

```diff
@@ -118,6 +133,19 @@ func message(r *http.Request) ([]byte, error) {
 	}
 }
 
+// BadRequestError is an error type that generates a 400 status code.
+type BadRequestError struct {
+	Message string
+}
+
+// Error returns the error message.
+func (e *BadRequestError) Error() string {
+	return e.Message
+}
+
+// StatusCode implements the kithttp StatusCoder interface
+func (e *BadRequestError) StatusCode() int { return http.StatusBadRequest }
```

Comment:
:+1: Once we move this to the monorepo, we'll probably be able to reuse the error type from the `service` package, but for now this LGTM!